### ISSUE TITLE
Variable is immediately overriden false positive

### ISSUE THREAD

**Issue by perk11** [kalessil/phpinspectionsea#778]
```
function getSite($user) {
    $site = $user->getOwnedSites()->first();
    if ($site === false) {
        throw new \Exception();
    }

    return $site;
}
```

This gives `$site is immediately overridden` warning

**Comment by kalessil** [comment]
Confirmed, this is a false-positive.

**Comment by kalessil** [comment]
@perk11 : cannot reproduce, what's plugin and IDE version do you use?

**Comment by kalessil** [comment]
Also: can you provide a screenshot, so I got ide which statement gets reported, please.

**Comment by perk11** [comment]
I'm sorry, this wasn't the full code causing the issue as I thought.
Here is the full version:
```
<?php

class A
{

    private function getTargetSite($user)
    {
        if ($user !== null) {
            $site = $user->getOwnedSites()->first();
            if ($site !== false) {
                return $site;
            }
        }

        $site = $user->getNetwork()->getSites()->first();
        if ($site === false) {
            throw new \Exception();
        }

        return $site;
    }


}
```
And a screenshot:
![image](https://user-images.githubusercontent.com/1924829/35693243-9e2af21a-078e-11e8-9221-bcd382904df5.png)
I am using IDEA 2017.3.3 with plugins Php Inspections (EA Extended) version 2.3.16.

**Comment by kalessil** [comment]
Perfect, thank you.

**Comment by kalessil** [comment]
Fixed
